docker/README.md

```diff
@@ -0,0 +1,23 @@
+# MLGO Development Dockerfiles
```

Comment:
a thought: should we name the directory `experimental`, and then there's even more hinting that this is... well... experimental/best effort support?

Reply:
Sounds like a decent idea to me. I moved the entire `docker/` directory to `experimental/docker/` in case we want to put anything else in there later and also to segment it a little bit better. Thanks for the suggestion.